server/src/main/java/org/opensearch/rest/action/admin/cluster/RestDecommissionAction.java

```diff
@@ -51,8 +52,11 @@ DecommissionRequest createRequest(RestRequest request) throws IOException {
         String attributeValue = request.param("awareness_attribute_value");
         // Check if we have no delay set.
         boolean noDelay = request.paramAsBoolean("no_delay", false);
-        if (noDelay) {
-            decommissionRequest.setNoDelay(noDelay);
+        decommissionRequest.setNoDelay(noDelay);
+
+        if (request.hasParam("delay_timeout")) {
+            TimeValue delayTimeout = request.paramAsTime("delay_timeout", DecommissionRequest.DEFAULT_NODE_DRAINING_TIMEOUT);
+            decommissionRequest.setDelayTimeout(delayTimeout);
```

Comment:
So technically its possible to have both params what happens if they are conflicting f.e no_delay = true and delay_timeout = 100s